### PR TITLE
Stream thickness

### PR DESCRIPTION
**Pull request by micmitch**
Added stream_thickness keyword argument to plotSlice and _plotImage2D functions so that it is possible to scale the thickness of streamlines to reflect the amplitude of the vector field. This functionality was added in a manner similar to the stream_threshold keyword.

stream_thickness keyword currently takes a float which acts as a scaling factor for the streamline thickness. Bounds are hardwired to fix the thickness of the 10% largest and smallest vector amplitudes. Provides good results with the DC current density plots I've made but could probably be generalized in the future for more flexibility.

### CODE REVIEW

**Comment by lheagy** [comment]
Thanks @micmitch: do you have a small example you could include in the discretize examples?

File: discretize/View.py [hunk 1]
```diff
@@ -494,6 +499,37 @@ def _plotImage2D(
...
+                upperBound = vecAmp_sort[-tenPercInd]
+
+                lowInds = np.where(vecAmp < lowerBound)
+                vecAmp[lowInds] = lowerBound
+
+                highInds = np.where(vecAmp > upperBound)
+                vecAmp[highInds] = upperBound
+
+                # Normalize amplitudes 0-1
+                # norm_thickness = (np.log10(vecAmp) - np.log10(vecAmp.min())) / (np.log10(vecAmp.max()) - np.log10(vecAmp.min()))
```
Comment (lheagy) [inline discretize/View.py]: would you mind cleaning out the commented-out line (523) if it is not needed? Thanks!

File: docs/examples/plot_streamThickness.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""
+Simple example to vary streamline thickness based on the vector amplitudes
```
Comment (lheagy) [inline docs/examples/plot_streamThickness.py]: Thanks for adding this @micmitch! Do you think we can shorten the title? (The whole title is included on the side-bar). What about
```
Plotting: Streamline thickness
=====================

A simple example to vary streamline thickness based on the vector amplitudes 

Author: `@micmitch <https://github.com/micmitch>`
```

File: discretize/View.py (hunk 1 above)
Comment (micmitch) [reply discretize/View.py]: Done!

File: docs/examples/plot_streamThickness.py (hunk 2 above)
Comment (micmitch) [reply docs/examples/plot_streamThickness.py]: Done!

**Comment by lheagy** [comment]
@micmitch: would you mind adding a code snippet showing example usage (and perhaps an image from your new example) to the description of this pull request? I will use your notes as the release notes when the version is bumped (e.g. https://github.com/simpeg/discretize/releases/tag/0.3.5)

**Comment by micmitch** [comment]
`# Plot secondary current densities for a pole transmitter in the ceiling of a tunnel with a conductive block 5 m above.

# Slice indices
xSliceInd = 49
ySliceInd = 30
zSliceInd = 30

# Axis limits
xlim = [-5, 55]
ylim = [-30, 30]
zlim = [-20, 20]

# Colorbar limits
clim = [5e-11, 4.0e-3]

# Source Index
srcInd = 5
src = srcList_Combined[srcInd]

# Current density plotting parameters 
label = 'Current Density ($A/m^2$)'
vType = 'F'
view = 'vec'
streamOpts = {'color':'w', 'density':2.0}
stream_threshold = 1e-8
pcolorOpts = {"cmap":"viridis"}

# Get secondary current density values from fields object
uTotal = fields_total[src,'j']
uPrim = fields_primary[src,'j']
u = uTotal - uPrim

# Slice to plot
sliceInd = xSliceInd
sliceNormal='X'

# Linear colormap
scale='linear'

# Plot colorbar
cb=True

# Set font sizes
labelsize = 30.
ticksize = 30.


# Setup figure
fig = plt.figure()
ax = plt.subplot(111)
fig.set_figheight(30)
fig.set_figwidth(15)

dat = mesh.plotSlice(u, ax=ax, normal=sliceNormal, ind=sliceInd, vType=vType, view=view, streamOpts=streamOpts, gridOpts={"color":"k", "alpha":0.1}, grid=True, range_x=ylim, range_y=zlim, clim=clim, stream_threshold=stream_threshold, stream_thickness=3)

# Plot outline of structures in true model slice
getBlkOutline(sigmaTrue_Block,mesh,sliceInd,normal=sliceNormal, ax=ax, lineWidth=3.0)

# 
ax.scatter(elecLocs[:,1],elecLocs[:,2], c='w', edgecolors='k', s=150)

if(isinstance(src, DC.SrcDC.Pole)):
    A = src.loc
    ax.plot(A[1],A[2], marker = 'o',color='red', markersize=15)
    yztextA1 = (A[1]-0.6,A[2] + 0.5)
    ax.annotate('A', xy=yztextA1, xytext=yztextA1,fontsize = labelsize, color='w', weight='bold')

elif(isinstance(src, DC.SrcDC.Dipole)):
    A = src.loc[1]
    B = src.loc[2]
    ax.plot(A[1],A[2], marker = 'o',color='red', markersize=15)
    ax.plot(B[1],B[2], marker = 'o',color='blue', markersize=15)

    yztextA1 = (A[1]-0.6,A[2] + 0.5)
    yztextB1 = (B[1]-0.6,B[2] + 0.5)

    ax.annotate('A', xy=yztextA1, xytext=yztextA1,fontsize = labelsize, color='w', weight='bold')
    ax.annotate('B', xy=yztextB1, xytext=yztextB1,fontsize = labelsize, color='w', weight='bold')

ax.set_title("N-S section at "+ "{0:.2f}".format(mesh.vectorCCx[sliceInd])+"m")
ax.set_xlim(ylim)
ax.set_ylim(zlim)

plt.gca().set_aspect('equal', adjustable='box')
plt.tight_layout()

if scale == "log":
    cbformat = "$10^{%1.1f}$"
elif scale == "linear":
    cbformat = "%.1e"
    
if cb:
    if clim is None:
        cmin, cmax = dat[0].get_clim()
    else:
        cmin, cmax = clim[0],clim[1]

    cb = add_colorbar(dat[0], format=cbformat, ax=ax, orientation="vertical",ticks = np.linspace(cmin, cmax, 5))
#         cb = plt.colorbar(dat[0], format=cbformat, ax = ax, orientation="vertical", fraction=0.0325, pad=0.04, ticks=np.linspace(cmin, cmax, 5))
    cb.ax.tick_params(labelsize=labelsize-4.0)
    cb.set_label(label, fontsize=labelsize)
    cb.set_clim([cmin, cmax])`

**Comment by micmitch** [comment]
[Js_SingleLinearArray_Top_Blk5m_8mElecBlkCenter_3x3Tunnel_X=25.pdf](https://github.com/simpeg/discretize/files/2659211/Js_SingleLinearArray_Top_Blk5m_8mElecBlkCenter_3x3Tunnel_X.25.pdf)

**Comment by micmitch** [comment]
Is this what you had in mind?

**Comment by lheagy** [comment]
The image looks great! I didn't mean for you to copy the whole example in, just a minimal part, but I can extract a subset

e.g. 
```python
mesh.plotSlice(
    u, ax=ax, normal='X', vType='F', view='vec', stream_threshold=1e-8, stream_thickness=3
)
```

**Pull request merged**